src/pages/index.js

```diff
@@ -20,88 +64,191 @@ const cardSection = new Section(
 imagePopup.setEventListeners();
 
 //  Initialize `UserInfo`
 const userInfo = new UserInfo({
   nameSelector: ".profile__title",
-  jobSelector: ".profile__description"
+  jobSelector: ".profile__description",
+  avatarSelector: ".profile__avatar"
 });
 
+function submitButtonTextChanger(button){
+  return button.textContent = 'Saving...'
+}
+
 const profilePopup = new PopupWithForm("#profileModal", (formData) => {
-  userInfo.setUserInfo(formData);
+  const originalText = profileSubmitButton.textContent;
+  
+  api.updateUserInfo(formData.title, formData.about) // sends to Api.js to save to database
+    .then((updatedData) => {
+      submitButtonTextChanger(profileSubmitButton);
+      console.log('Changed profile submit button to saving...')
+      userInfo.setUserInfo(formData);
+      profilePopup.close(); 
+    })
+    .catch((err) => {
+      console.error("Error updating profile:", err);
+    })
+    .finally(()=>{
+      profileSubmitButton.textContent = originalText;
+      console.log('changing prof submit button text back to original')
+    })
 });
+
 profilePopup.setEventListeners();
 
+
 const addCardPopup = new PopupWithForm("#modal__add", (formData) => {
+const originalText = addCardPopupSubmit.textContent;
+
   const newCardData = {
     name: formData.name.trim(),
     link: formData.url.trim(),
   };
 
   if (newCardData.name && newCardData.link) {
-    cardSection.addItem(createCard(newCardData));
-
-    // Only reset form on submit
-    document.querySelector("#add__modal_form").reset();
-
-    // Close popup after adding a card
-    addCardPopup.close();
+    api.updateNewCards(newCardData) 
+      .then((savedCard) => {
+        submitButtonTextChanger(addCardPopupSubmit);
+        // Ensure the savedCard has all necessary properties
+        const fullCardData = {
+          ...savedCard,
+          likes: [], // Initialize empty likes array
+          _id: savedCard._id, // Ensure _id is present
+        };
+
+        const cardElement = createCard(fullCardData)
+        cardSection.addItem(cardElement); 
+        addModalForm.reset();
+        addCardPopup.close(); 
+      })
+      .catch((err) => {
+        console.error("Error adding new card:", err);
+      })
+      .finally(()=>{
+        addCardPopupSubmit.textContent = originalText;
+        //to test the submitButtonChanger please 
+        //remove the code on line 160 and you'll see the text is changed 
+        console.log('changing add submit button text back to original')
+      })
   }
 });
+addCardPopup.setEventListeners();
 
+const avatarPopup = new PopupWithForm("#avatar__modal", (formData) => {
+  console.log("Avatar Form Data:", formData);
+  const originalText = avatarPopupSubmit.textContent;
+  
+
+  api.updateAvatar(formData.avatar_url) 
+    .then((updatedData) => {
+      userInfo.setAvatar(updatedData); // Updates avatar 
+      submitButtonTextChanger(avatarPopupSubmit);
+      document.querySelector("#avatar__modal_form").reset(); 
```

Comment:
`document.querySelector("#avatar__modal_form")`


You search the form 2 times. Here and in line 89. Please, find the `avatar` form only once 


```js

const avatarForm = document.querySelector("#avatar__modal_form")



```


 Then use it for the validator

```js

 const avatarValidator = new FormValidator(config, avatarForm);


```


And use it here


```js
      avatarForm.reset(); 


```